lib/bigtable/row.js

```diff
@@ -0,0 +1,745 @@
+ *
+ * //-
+ * // You can also create a rule that will append data to an existing value.
+ * // If the targeted cell is unset, it will be treated as a containing an
+ * // empty string.
+ * //-
+ * row.createRules([
+ *   {
+ *     column: 'follows:alincoln',
+ *     append: ' Honest Abe!'
+ *   }
+ * ], callback);
+ */
+Row.prototype.createRules = function(rules, callback) {
+  rules = arrify(rules).map(function(rule) {
+    var column = Mutation.parseColumnName(rule.column);
+    var ruleData = {
+      familyName: column.family,
+      columnQualifier: Mutation.convertToBytes(column.qualifier)
+    };
+
+    if (rule.append) {
+      ruleData.appendValue = Mutation.convertToBytes(rule.append);
+    }
+
+    if (rule.increment) {
+      ruleData.incrementAmount = rule.increment;
+    }
+
+    return ruleData;
+  });
+
+  var grpcOpts = {
+    service: 'BigtableService',
+    method: 'readModifyWriteRow'
+  };
+
+  var reqOpts = {
+    tableName: this.parent.id,
+    rowKey: Mutation.convertToBytes(this.id),
+    rules: rules
+  };
+
+  this.request(grpcOpts, reqOpts, callback);
+};
+
+/**
+ * Mutates a row atomically based on the output of a filter. Depending on
+ * whether or not any results are yielded, either the `onMatch` or `onNoMatch`
+ * callback will be executed.
+ *
+ * @param {module:bigtable/filter} filter - Filter ot be applied to the contents
+ *     of the row.
+ * @param {?object[]} onMatch - A list of entries to be ran if a match is found.
+ * @param {object[]=} onNoMatch - A list of entries to be ran if no matches are
+ *     found.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this
+ *     request.
+ * @param {boolean} callback.matched - Whether a match was found or not.
+ *
+ * @example
+ * var callback = function(err, matched) {
+ *   if (!err) {
+ *     // `matched` will let us know if a match was found or not.
+ *   }
+ * };
+ *
+ * var filter = [
+ *   {
+ *     family: 'follows'
+ *   }, {
+ *     column: 'alincoln',
+ *   }, {
+ *     value: 1
+ *   }
+ * ];
+ *
+ * var entries = [
+ *   {
+ *     method: 'insert',
+ *     data: {
+ *       follows: {
+ *         jadams: 1
+ *       }
+ *     }
+ *   }
+ * ];
+ *
+ * row.filter(filter, entries, callback);
+ *
+ * //-
+ * // Optionally, you can pass in an array of entries to be ran in the event
+ * // that a match is not made.
+ * //-
+ * row.filter(filter, null, entries, callback);
+ */
+Row.prototype.filter = function(filter, onMatch, onNoMatch, callback) {
+  var grpcOpts = {
+    service: 'BigtableService',
+    method: 'checkAndMutateRow'
+  };
+
+  if (is.function(onNoMatch)) {
+    callback = onNoMatch;
+    onNoMatch = [];
+  }
+
+  var reqOpts = {
+    tableName: this.parent.id,
+    rowKey: Mutation.convertToBytes(this.id),
+    predicateFilter: Filter.parse(filter),
+    trueMutations: createFlatMutationsList(onMatch),
+    falseMutations: createFlatMutationsList(onNoMatch)
```

Comment:
Woops, yeah I messed up with the `.prop` thing. I would still use the one liner instead of the inner fn (just arrify earlier / split arr mods to new lines), but your call.